ecommerce/extensions/api/serializers.py

```diff
@@ -827,13 +829,30 @@ def validate_products(self, products):
 
         return products
 
-    def _update_mobile_seats(self, course):
+    def _update_or_create_mobile_seats(self, course):
         certificate_type_query = Q(attributes__name='certificate_type',
                                    attribute_values__value_text=CertificateType.VERIFIED)
         mobile_query = Q(stockrecords__partner_sku__contains='mobile')
         seat_products = course.seat_products
         mobile_seats = seat_products.filter(certificate_type_query & mobile_query)
         web_seat = seat_products.filter(certificate_type_query & ~mobile_query).first()
+        if mobile_seats:
+            self._update_mobile_seats(mobile_seats, web_seat, course)
+        else:
+            logger.info("Creating mobile seats for course [%s]", course.id)
+            create_mobile_seat(ANDROID_SKU_PREFIX, web_seat)
```

Comment:
Wrap `create_mobile_seat` calls in try-except to catch specific failures (e.g., Android or iOS creation).
something like this
```suggestion

try:
        create_mobile_seat(ANDROID_SKU_PREFIX, web_seat)
    except Exception as e:
        logger.error("Failed to create Android mobile seat for course [%s]: %s", course.id, str(e))
        return
```

Reply:
@jawad-khan Please address this comment also